### ISSUE TITLE
Adding both chest- and tank-carts to a locomotive possible. Bug or feature?

### ISSUE THREAD

**Issue by wshadow** [Railcraft/Railcraft#635]
Railcraft 9.6.1.15
No other mods besides Forge.
Singleplayer test world.

Since "Steam Locomotives can now pull fuel and water from attached carts." I wanted to try and add both types. According to the changelog this is not possible, and when trying with just one cart of each type and one locomotive, this seems to be true. However, when you add a second locomotive to the train this works just fine. I took some screenshots, see https://drive.google.com/open?id=0Bz_fgROOZk6VfnpWVXc2ZVl0SzgyNEZ5UHhkdUMyRnl2ZGx6dUlEQXpfTXFNckhNcWJxUWM&authuser=0

To make it clear:
- The setup with just one locomotive doesn't work. linking the chest cart fails
- The setup with two locomotives works. Both locomotives pull water and fuel from the respective carts.

While I would consider this to be a feature; The changelog and the fact that this doesn't work with a single locomotive in the middle, suggests to me, that this is not intended.

**Comment by CovertJaguar** [comment]
Amusing, I didn't think of that. =P

**Comment by liach** [comment]
@CovertJaguar Is this still a bug?

**Comment by CovertJaguar** [comment]
Yes, I haven't touched it.

**Comment by liach** [comment]
Blocked by #1372

**Comment by Lilithvia** [comment]
I think locomotives should be able to pull from both cargo/chest carts and tank carts at the same time.

**Comment by liach** [comment]
Each type of minecart has their own transfer limitations, e.g. cargos cannot transfer fluid or items of different type, tanks can only transfer fluid of its type, and chests not compatible with fluids. The locomotive was able to work in the second case solely because both water and fuel can go through locomotives.

**Comment by liach** [comment]
Or we may add upgrades to allow certain carts transfer more than one type of stuff at a time

**Comment by Lilithvia** [comment]
Those kinds of upgrades, or a special cart just for locomotives to pull from (Maybe named "tender cart" like IRL steam locomotives pull both fuel and water from) would be great.

**Comment by Forecaster** [comment]
That has been discussed in the past. It would have limited capacity though.

**Comment by Lilithvia** [comment]
I've never really played with cargo carts. I might have to just to give my suggestion for the capacity of tender carts,

**Comment by Lilithvia** [comment]
Tender discussion should have its own issue.

**Comment by liach** [comment]
@Lilithvia Next time when you comment, try to express all your ideas in one comment. Some people receive emails from github, and comment like yours may spam their email.

Edit: Feel free to make an issue about tender cart

**Comment by liach** [comment]
After looking, I'd say this is working as intended. I'd personally remove the linking restriction posed by locos.

**Comment by Generalcamo** [comment]
This is definitely a bug. Only a cart behind the locomotive should be considered for fuel linkage.

**Comment by liach** [comment]
Fixed in https://github.com/Railcraft/Railcraft/commit/d397f4b7586af8ac3a9194b77d2fd9aa09952429#diff-1c0b14726287c5acff07e8aea238cd1eR695
